### PR TITLE
Detect and add logging for done event loop

### PR DESCRIPTION
**Pull request by alex-deaconu**
If a Widget sends done more than once a second, record it
If it has done it 10 times in a row, log to BQ
Ignore it after

Adding console entry for BQ logging

### CODE REVIEW

**Comment by alex-deaconu** [comment]
Logging to BQ if the loop happens more than 10 times for a particular widget instance. Also, not responding to the done event more than once a second.

To clarify, this doesn't resolve the issue with the Facebook Widget, but at least it will allow us to detect this incorrect behaviour.

Replicate at (takes about 10-20 mins for the event to be logged): http://done-loop-dot-rvaviewer-test.appspot.com/?type=schedule&id=6d3b9c16-c3f4-4bcc-8c34-17d782dbd997&showui=false

Query to validate: `select * FROM [Viewer_Events.events20160711] WHERE event = 'Widget Done Loop' order by ts desc`

@ezequielc please review. @tejohnso FYI. Thanks!

File: circle.yml [hunk 1]
```diff
@@ -5,7 +5,7 @@ dependencies:
 checkout:
   post:
     # Update environment depending URLs
-    - if [ $CIRCLE_BRANCH == "master" ]; then export DEPLOYENV=prod; else export DEPLOYENV=staging; fi; cat src/main/java/com/risevision/viewer/client/info/Global.java | awk -v DEPLOYENV=$DEPLOYENV 'BEGIN{OFS=FS="="}/String ENV/{$2=" \"" DEPLOYENV "\";"}1' | awk 'gsub("{dateTime}", strftime("%Y%m%d%H%M"))1' > Global.tmp
+    - if [ $CIRCLE_BRANCH == "master" ]; then export DEPLOYENV=prod; else export DEPLOYENV=prod; fi; cat src/main/java/com/risevision/viewer/client/info/Global.java | awk -v DEPLOYENV=$DEPLOYENV 'BEGIN{OFS=FS="="}/String ENV/{$2=" \"" DEPLOYENV "\";"}1' | awk 'gsub("{dateTime}", strftime("%Y%m%d%H%M"))1' > Global.tmp
```
Comment (ezequielc) [inline circle.yml]: @alex-deaconu Curious, why did you need to change the environment here? Thanks

**Comment by ezequielc** [comment]
@alex-deaconu Code looks good. Just one question. Thanks

File: src/main/java/com/risevision/viewer/client/controller/ViewerWidgetController.java [hunk 2]
```diff
@@ -153,6 +158,24 @@ public void setError(String presFrame, String reason) {
...
+			if (lastDone != 0 && new Date().getTime() - lastDone < 1000) {				
+				lastDone = new Date().getTime();
+
+				// -1 indicates already logged
+				if (doneCounter != -1) {
+					doneCounter++;
+					
+					if (doneCounter >= 10) {
+						ViewerHtmlUtils.logExternalMessage("Widget Done Loop", playlistItem.getObjectRef());
+
```
Comment (tejohnso) [inline src/main/java/com/risevision/viewer/client/controller/ViewerWidgetController.java]: @alex-deaconu Can you change to "widget done loop" for consistency?

Comment (alex-deaconu) [inline src/main/java/com/risevision/viewer/client/controller/ViewerWidgetController.java]: @tejohnso corrected. Thanks for the heads up!

File: circle.yml (hunk 1 above)
Comment (alex-deaconu) [inline circle.yml]: @ezequielc @tejohnso sorry forgot to mention this. The Schedule I used for testing works only in Production (the Widget is not registered on the Test server), so I changed the staging to `prod` for testing.

I've just reverted the change however. Thanks.

**Comment by tejohnso** [comment]
:+1:

**Comment by ezequielc** [comment]
@alex-deaconu 👍
